### DISCUSSION TITLE
http://example.com:8080 - https://example.com:8080

### DISCUSSION THREAD

**Discussion by VlaK0r** [dani-garcia/vaultwarden#3697] [Q&A]
Hello!
Thank you for your work.
There is a domain with a certificate https://example.com
Installed vaultwarden on the server via docker. docker run -d --name vaultwarden -v /vw-data/:/data/ --restart unless-stopped -p 8080:80 vaultwarden/server:latest
It opens at http://example.com:8080
How to make it open at https://example.com:8080 or https://example.com/vaultwarden

**Comment by Masgalor** [comment]
There is a wiki page about enabling https.
https://github.com/dani-garcia/vaultwarden/wiki/Enabling-HTTPS

**Reply by VlaK0r** [reply]
I don't understand this well.
I've already read many pages and still can't figure out how to do it.
Help please.

**Reply by Masgalor** [reply]
Well, I can't tell you anything that isn't already explained on that page.

Maybe you could explain what you already tried and why this did not work out?

**Reply by VlaK0r** [reply]
I got certificates for nginx. Website https://example.com now it works.
I'm trying to configure nginx reverse proxy. But nothing happens.

**Reply by Masgalor** [reply]
Okay.
So you already have some website running under your domain and now you want to get vaultwarden running under a subdomain or subpath like `vaultwarden.example.com`  or `example.com/vaultwarden` ?

**Reply by VlaK0r** [reply]
Sorry for the long wait. Fatigue won me over.
example.com/vaultwarden

**Reply by VlaK0r** [reply]
Or vaultwarden.example.com which is easier?

**Reply by Masgalor** [reply]
I wouldn't say one is more easy than the other, but using a subdomain needs less configuration.
Fortunately there are wiki pages for both scenarios.

Subdomain:
https://github.com/dani-garcia/vaultwarden/wiki/Proxy-examples -> Nginx - v1.29.0+ (by BlackDex)

Subpath:
https://github.com/dani-garcia/vaultwarden/wiki/Using-an-alternate-base-dir
https://github.com/dani-garcia/vaultwarden/wiki/Proxy-examples -> Nginx with sub-path - v1.29.0+ (by BlackDex)

**Reply by VlaK0r** [reply]
I decided to do it through Apache after all.
Here is according to this instruction.
`Apache in a sub-location (by ss89)

Modify your docker start-up to include the sub-location.
; Add the sub-location! Else this will not work!
DOMAIN=https://$hostname.$domainname/$sublocation/
Ensure you have the websocket proxy module loaded somewhere in your apache config. It can look something like:

LoadModule proxy_wstunnel_module modules/mod_proxy_wstunnel.so`
On some OS's you can use a2enmod, for example with: a2enmod proxy_wstunnel and a2enmod proxy_http.

<VirtualHost *:443>
    SSLEngine on
    ServerName $hostname.$domainname

    SSLCertificateFile ${SSLCERTIFICATE}
    SSLCertificateKeyFile ${SSLKEY}
    SSLCACertificateFile ${SSLCA}
    ${SSLCHAIN}

    ErrorLog \${APACHE_LOG_DIR}/error.log
    CustomLog \${APACHE_LOG_DIR}/access.log combined

    <Location /$sublocation/> #adjust here if necessary
        RewriteEngine On
        RewriteCond %{HTTP:Upgrade} =websocket [NC]
        RewriteRule /notifications/hub(.*) ws://<SERVER>:3012/$1 [P,L]
        ProxyPass http://<SERVER>:80/$sublocation/

        ProxyPreserveHost On
        RequestHeader set X-Real-IP %{REMOTE_ADDR}s
    </Location>
</VirtualHost>`

**Reply by VlaK0r** [reply]
But I don't understand where to insert all these records

**Reply by LanceMcCarthy** [reply]
The virtual host goes into it's own file inside `sites-enabled` and `sites-available`. You woukld have a file for each site, or edit the main one (not really recommended). Here's a tutorial => https://www.digitalocean.com/community/tutorials/how-to-set-up-apache-virtual-hosts-on-ubuntu-20-04

## GUI - Only Option
That said, I thought it might be worth showing you an easier way to manage reverse proxy routes and managing LetsEncrypt certs. Take a look at [nginxproxymanager](https://nginxproxymanager.com). It's GUI based where you can easily setup all the proxy routes with automatic LetsEncrypt renewals for the certs.

To set it up, it's literally a simple docker compose file, once started navigate to the admin portal. See => https://nginxproxymanager.com/guide/#quick-setup

> I don't work for nginx or anything, this is just advice from my own experience after hosting my own homelab services for many years. It's really nice to be able to go to a portal to add/remove routes and to have it automatically renew my certs. It JustWorks™

In the web GUI., it's fantastic. You can add routes, manage SSL certifications and more. Here are some screenshots => https://nginxproxymanager.com/screenshots/

>note You can't run nginx and apache together side-by-side, you'd pick one or the other. But in this case, since nginxproxymanager runs inside a docker container, you don't need to worry about having apache in the host environment! :D

Later on, after you get used to how the concept of reverse proxy and the routes are configured, then you can wade into the more advanced areas and play with apache configs

About subdomain vs route. I would personally recommend using a subdomain as a beginner (myvault.domain.com)... yes you do need to add a record to your DNS, but that only takes seconds. For me, it's worth the clear differentiation between what each subdomain does.
